### PR TITLE
[AMORO-1765]: Add metric collection API

### LINKED ISSUE #1765: [Improvement]: Add metric collection API
### Search before asking

- [X] I have searched in the [issues](https://github.com/NetEase/arctic/issues?q=is%3Aissue) and found no similar issues.

### What would you like to be improved?

Add metric collection API

### How should we improve?

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Subtasks

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by hameizi**
## Why are the changes needed?
resolve #1765 

## Brief change log

Add metric interface

## How was this patch tested?
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] Run test locally before making a pull request

## Documentation

  - Does this pull request introduce a new feature? (yes)
  - If yes, how is the feature documented? (not documented)

### CODE REVIEW

File: ams/api/src/main/java/com/netease/arctic/ams/api/metrics/IcebergCommitReport.java [hunk 1]
```diff
@@ -20,153 +20,230 @@
...
+    this.snapshotId = snapshotId;
+    this.sequenceNumber = sequenceNumber;
+    this.operation = operation;
+    this.metadata = metadata;
+  }
 
   @Override
-  default String name() {
+  public String name() {
     return "iceberg_commit_report";
```
Comment (zhoujinsong) [inline ams/api/src/main/java/com/netease/arctic/ams/api/metrics/IcebergCommitReport.java:102]: I recommend adding a static variable for the name too.

File: ams/api/src/main/java/com/netease/arctic/ams/api/metrics/IcebergCommitReport.java [hunk 2]
```diff
@@ -20,153 +20,230 @@
...
 
   @TaggedMetrics.Tag(name = TABLE_NAME)
-  String tableName();
+  public String tableName() {
+    return this.tableName;
+  }
 
   @TaggedMetrics.Tag(name = SNAPSHOT_ID)
-  long snapshotId();
+  public long snapshotId() {
```
Comment (zhoujinsong) [inline ams/api/src/main/java/com/netease/arctic/ams/api/metrics/IcebergCommitReport.java:111]: I don't think the Snapshot ID is a tag for the commit metric, maybe a gauge metric value?

File: ams/api/src/main/java/com/netease/arctic/ams/api/metrics/IcebergCommitReport.java [hunk 3]
```diff
@@ -20,153 +20,230 @@
...
+    return this.snapshotId;
+  }
 
   @TaggedMetrics.Tag(name = SEQUENCE_NUMBER)
-  long sequenceNumber();
+  public long sequenceNumber() {
```
Comment (zhoujinsong) [inline ams/api/src/main/java/com/netease/arctic/ams/api/metrics/IcebergCommitReport.java:116]: The same with Snapshot ID.

File: ams/api/src/main/java/com/netease/arctic/ams/api/metrics/IcebergCommitReport.java [hunk 4]
```diff
@@ -20,153 +20,230 @@
...
 
   @TaggedMetrics.Tag(name = OPERATION)
-  String operation();
+  public String operation() {
+    return this.operation;
+  }
 
   @TaggedMetrics.Tag(name = METADATA)
-  Map<String, String> metadata();
+  public Map<String, String> metadata() {
```
Comment (zhoujinsong) [inline ams/api/src/main/java/com/netease/arctic/ams/api/metrics/IcebergCommitReport.java:126]: The same with Snapshot ID.

File: ams/api/src/main/java/com/netease/arctic/ams/api/metrics/IcebergCommitReport.java (hunk 2 above)
Comment (huyuanfeng2018) [reply ams/api/src/main/java/com/netease/arctic/ams/api/metrics/IcebergCommitReport.java:111]: +1

**Review by zhoujinsong — APPROVED** [review]

**Pull request merged**